### PR TITLE
Fix warning about the use of tmpnam

### PR DESCRIPTION
**Pull request by Kgeorgouli**

### CODE REVIEW

File: src/utils/generate_cxx_code.cpp [hunk 1]
```diff
@@ -1302,11 +1302,19 @@ const std::string  wcs::generate_cxx_code::generate_code(
 
   const char * Real = generate_cxx_code::basetype_to_string<reaction_rate_t>::value;
 
-  char * pointerFilename;
-  pointerFilename = tmpnam (NULL);
-  //std::cout << "name file:" << pointerFilename << ".cc" << "\n";
+  int filename_sz = 25;
```
Comment (JaeseungYeom) [inline src/utils/generate_cxx_code.cpp]: Move this line to wcs_types.hpp as `constexpr size_t wcs_gen_path_max = 25u;` with a comment line above explaining what it is.
And replace `filename_sz` with `wcs_gen_path_max`

**Review by JaeseungYeom — APPROVED** [review]

File: src/utils/generate_cxx_code.cpp [hunk 2]
```diff
@@ -1302,11 +1302,19 @@ const std::string  wcs::generate_cxx_code::generate_code(
...
+  char nameBuff[filename_sz];
+  int fd = -1;
+  strncpy(nameBuff,"/tmp/wcs_generatedXXXXXX",filename_sz);
```
Comment (JaeseungYeom) [inline src/utils/generate_cxx_code.cpp]: `wcs_generatedXXXXXX`->`wcs_generated_XXXXXX`

Comment (Kgeorgouli) [reply src/utils/generate_cxx_code.cpp]: Done

File: src/utils/generate_cxx_code.cpp (hunk 1 above)
Comment (Kgeorgouli) [reply src/utils/generate_cxx_code.cpp]: Done

**Pull request merged**